#1111: Paused enrollment indicator

andymikulski opened the pull request:
Fixes #1111 (mostly)

Ticket #1111 requests an indicator on the recipe details page for 'paused enrollment', however this data already exists on the page: 
![screen shot 2017-12-18 at 12 41 15 pm](https://user-images.githubusercontent.com/2767162/34127993-699be8c0-e3fc-11e7-90b8-0b530dd5f3b9.png)

The ticket requested adding this info to the listing page, which this PR does:
![screen shot 2017-12-18 at 2 03 39 pm](https://user-images.githubusercontent.com/2767162/34128020-7d573e6e-e3fc-11e7-81b0-dde2c0233b59.png)

This should, for the most part, close #1111, as the 'paused' status is now available on the `listing`, `details`, and `edit` pages.

mythmon commented:
@MattGrimes do these screenshots look good to you?

mythmon commented on recipe-server/client/control/components/common/EnrollmentStatus.js:
```diff
@@ -0,0 +1,83 @@
+import { Link } from 'redux-little-router';
+import { connect } from 'react-redux';
+
+@connect((state, { record }) => {
```
This isn't actually using `state` anywhere, as far as I can see. We probably don't want to `@connect` things that don't read from Redux. I would expect things like `isEnabled` and `isPaused` to be calculated in `render`, since they are derived from this component's props.

andymikulski replied:
That's actually how it started out, but for testing purposes I felt it was more straightforward to expose the `isEnabled`/`isPaused` props. I can disconnect it.

mythmon replied:
I think using `isEnabled`/`isPaused` directly in tests skips out on testing an important bit of logic, that is, how those fields are determined.

andymikulski commented:
CI error is due to tests using `.find(Icon)`. I should probably figure out why that works locally for me but not on Circle.

mythmon commented:
> CI error is due to tests using .find(Icon). I should probably figure out why that works locally for me but not on Circle.

I imagine that you and Circle are using different versions of Firefox. Could that do it? It may also be a Linux vs OSX thing?

andymikulski commented:
I've removed the `connect` bits, and refactored some duplicate code.

I have also decided to remove the filtering for the `paused` column after hitting some friction getting it to work consistently. I imagine we can revisit this piece when we implement the back end functionality.

Ready for another pass!

mythmon commented:
This looks good now, thanks for the changes!

bors r+

mythmon commented:
I'm not sure why the test timed out, but it also has test failures like this that need fixing:

```
✗ should print `paused` if the recipe is enabled and paused
    Expected '<Connect(L) />' to contain 'Paused'.
```

mythmon commented:
bors r+
